### PR TITLE
✨ #65979 Migrate set initial directory

### PR DESCRIPTION
**Pull request by gantelo**
To be cross-reviewed by team

### CODE REVIEW

File: packages/file_selector/file_selector_windows/lib/src/dart_file_selector_api.dart [hunk 1]
```diff
@@ -109,6 +110,34 @@ class DartFileSelectorAPI extends FileDialog {
...
+      final Pointer<GUID> guid = GUIDFromString(IID_IShellItem);
+      final Pointer<Pointer<COMObject>> dirPath = arena<Pointer<COMObject>>();
+      hResult = SHCreateItemFromParsingName(
+          TEXT(initialDirectory), nullptr, guid, dirPath);
+
+      if (FAILED(hResult)) {
+        throw WindowsException(hResult);
+      }
+
+      hResult = dialog.setFolder(dirPath.value);
```
Comment (juandausa) [inline packages/file_selector/file_selector_windows/lib/src/dart_file_selector_api.dart:131]: Have you seen the abstraction we are using to interact with Dialogs?

File: packages/file_selector/file_selector_windows/test/dart_file_selector_api_test.dart [hunk 2]
```diff
@@ -193,6 +194,52 @@ void main() {
...
+    test('setInitialDirectory should success when initialDirectory is valid',
+        () {
+      expect(successReturnValue,
+          api.setInitialDirectory(defaultReturnValue, defaultPath, dialog));
+    });
+
+    test('setInitialDirectory should throw Error 0x80070002 when initialDirectory is an inexistent path',
+        () {
+      expect(
+          () => api.setInitialDirectory(defaultReturnValue, r'C:\INEXISTENT_DIR', dialog),
```
Comment (juandausa) [inline packages/file_selector/file_selector_windows/test/dart_file_selector_api_test.dart:229]: Can we guarantee that a path is available? What if I run this test suite on Linux?

File: packages/file_selector/file_selector_windows/lib/src/dart_file_selector_api.dart (hunk 1 above)
Comment (gantelo) [reply packages/file_selector/file_selector_windows/lib/src/dart_file_selector_api.dart:131]: Oh I've just seen that way. Do you mean to move the setFolder to fileselectorapi.setInitialDirectory ?

File: packages/file_selector/file_selector_windows/test/dart_file_selector_api_test.dart (hunk 2 above)
Comment (gantelo) [reply packages/file_selector/file_selector_windows/test/dart_file_selector_api_test.dart:229]: Not really, wouldn't that be like running mac tests on windows?

Comment (juandausa) [reply packages/file_selector/file_selector_windows/test/dart_file_selector_api_test.dart:229]: jajaja is true, but what about if `c:` is unavailable? My concern is related to the CI sever running these tests.

Comment (gantelo) [reply packages/file_selector/file_selector_windows/test/dart_file_selector_api_test.dart:229]: Maybe we can go with just 'INEXISTENT_DIR' to be sure 🤔, I'll test how it runs in windows

Comment (gantelo) [reply packages/file_selector/file_selector_windows/test/dart_file_selector_api_test.dart:229]: just tested it and it runs smoothly, so I'll change it to that

**Review by eugerossetto — APPROVED** [review]
LGTM

**Pull request merged**